ddtrace/internal/service.py

```diff
@@ -48,9 +62,94 @@ def _start(self):
         """
 
     def stop(self):
+        # type: () -> None
         """Stop the service."""
         self.status = ServiceStatus.STOPPED
 
     @staticmethod
-    def join(timeout=None):
+    def join(
+        timeout=None,  # type: float
+    ):
+        # type: (...) -> typing.Optional[float]
         """Join the service once stopped."""
+
+    def copy(self):
+        # type: () -> Service
+        return attr.evolve(self)
+
+
+class ServiceProcess(object):
```

Comment:
I'm not quite happy with this name as, technically, this is not a process. I would suggest renaming it to something like `ForksafeService`

Reply:
Sure I can rename it. I'll wait a bit in case other folks have different name ideas. 😄

Reply:
I don't have a strong opinion. ForksafeService is pretty explicit which is good :)